developers/weaviate/current/configuration/authentication.md

```diff
@@ -116,29 +116,45 @@ services:
+### Weaviate OpenID endpoint
 
 If you have authentication enabled, you can obtain Weaviate's OIDC configuration from the following endpoint:
 
 ```bash
 $ curl [WEAVIATE URL]/v1/.well-known/openid-configuration
 ```
 
-### How to use
+## How to use
 
 OIDC authentication requires obtaining a valid token from the token issuer so that it can be sent in the header of any request to Weaviate. This applies to both REST and GraphQL requests.
 
-The OIDC standard allows for many different methods *(flows)* of obtaining a token. The appropriate method can vary depending on your situation, including configurations at the token issuer, and your requirements. 
+The OIDC standard allows for many different methods *(flows)* of obtaining a token. The appropriate method can vary depending on your situation, including configurations at the token issuer, and your requirements.
 
 While it is outside the scope of our documentation to cover every OIDC authentication flow, some possible options are to:
+- Use the `client credentials flow` for machine-to-machine authorization. (Note that this will authorize an app, rather than a particular user.)
 - Use the `resource owner password flow` for trusted applications. 
   - You can use Weaviate Cloud Services / WCS (`https://auth.wcs.api.semi.technology/`) as the token issuer using the `resource owner password flow`.
   - Weaviate's Python client directly supports this method.
 - Or use the `hybrid flow` if Microsoft Azure AD is your token issuer or if you would like to prevent exposing passwords.
 
 We outline the steps below for both methods of obtaining a token.
 
-#### Resource owner password flow 
+### Client credentials grant
+
+Note: At the time of writing (December 2022), GCP does not support client credentials grant flow.
+
+1. Send a GET request to `[WEAVIATE_URL]/v1/.well-known/openid-configuration` to fetch Weaviate's OIDC configuration (`wv_oidc_config`)
+2. Parse the `clientId` and `href` from `wv_oidc_config`
+3. Send a GET request to `href` to fetch the token issuer's OIDC configuration (`token_oidc_config`)
+4. Check that the `token_endpoint_auth_methods_supported` response includes a `client_secret_<specific_method>`. We will base the remaining instruction based on `client_secret_post` being enabled.
+5. Send a POST request to the `token_endpoint` of `token_oidc_config` with the body: 
+  - `{"grant_type": "client_credentials", "client_id": client_id, "client_secret": [CLIENT_SECRET], "scope": [SCOPE]}`. 
+  - Where `[CLIENT_SECRET]` is replaced with the actual client secret, and `[SCOPE]` is replaced with the appropraite value for the provider.
+    - For Okta, the `[SCOPE]` value should be the one set in the `authorization server` configuration.
+    - For Azure AD, the `[SCOPE]` value should be `[CLIENT_ID]/.default` where `[CLIENT_ID]` is replaced with the actual client id value.
+6. Parse the response (`token_resp`), and look for `access_token` in `token_resp`. This is your Bearer token.
```

Comment:
I'm unsure if this is necessary here.

We have two methods for everyone to authenticate manually which should be good enough and hopefully we can let the clients handle the client credentials

Reply:
Okay. I have removed the entire section on manually obtaining a token with client credentials flow.

Reply:
Maybe add it to our internal confluence page, so the work is not lost?

Reply:
Great idea. Done!